tests/cstor/invalidconfig/invalidconfig_test.go

```diff
@@ -0,0 +1,231 @@
+				Build().Object
+
+			By("creating storagepoolclaim")
+			_, err = ops.SPCClient.Create(spcObj)
+			Expect(err).To(BeNil(), "while creating spc", spcName)
+
+			By("verifying healthy csp count as 0")
+			cspCount := ops.GetHealthyCSPCount(spcName, cstor.PoolCount)
+			Expect(cspCount).To(Equal(0), "while checking cstorpool health status")
+
+		})
+	})
+
+	When("creating storagepoolclaim with invalid pool type", func() {
+		It("should not create any cstorpool", func() {
+
+			By("building spc object")
+			spcObj = spc.NewBuilder().
+				WithName(spcName).
+				WithDiskType(string(apis.TypeSparseCPV)).
+				WithMaxPool(cstor.PoolCount).
+				WithOverProvisioning(false).
+				WithPoolType(string("invalid-pool-type")).
+				Build().Object
+
+			By("creating storagepoolclaim")
+			_, err = ops.SPCClient.Create(spcObj)
+			Expect(err).To(BeNil(), "while creating spc", spcName)
+
+			By("verifying healthy csp count as 0")
+			cspCount := ops.GetHealthyCSPCount(spcName, cstor.PoolCount)
+			Expect(cspCount).To(Equal(0), "while checking cstorpool health status")
+
+		})
+	})
+
+	When("creating storagepoolclaim with invalid pool count", func() {
+		It("should not create any cstorpool", func() {
+
+			By("building spc object")
+			spcObj = spc.NewBuilder().
+				WithName(spcName).
+				WithDiskType(string(apis.TypeSparseCPV)).
+				WithMaxPool(-1).
+				WithOverProvisioning(false).
+				WithPoolType(string(apis.PoolTypeStripedCPV)).
+				Build().Object
+
+			By("creating storagepoolclaim")
+			_, err = ops.SPCClient.Create(spcObj)
+			Expect(err).To(BeNil(), "while creating spc", spcName)
+
+			By("verifying healthy csp count as 0")
+			cspCount := ops.GetHealthyCSPCount(spcName, cstor.PoolCount)
+			Expect(cspCount).To(Equal(0), "while checking cstorpool health status")
+
+		})
+	})
+
+})
+
+var _ = Describe("[cstor] TEST INVALID STORAGECLASS", func() {
+	var (
+		err                   error
+		pvcName               = "cstor-volume-claim"
+		openebsCASConfigValue = `
+- name: ReplicaCount
+  value: $count
+- name: StoragePoolClaim
+  value: test-cstor-provision-sparse-pool-auto
+`
+	)
+
+	BeforeEach(func() {
+		When(" creating a cstor based volume", func() {
+
+			By("building spc object")
+			spcObj = spc.NewBuilder().
+				WithName(spcName).
+				WithDiskType(string(apis.TypeSparseCPV)).
+				WithMaxPool(cstor.PoolCount).
+				WithOverProvisioning(false).
+				WithPoolType(string(apis.PoolTypeStripedCPV)).
+				Build().Object
+
+			By("creating storagepoolclaim")
+			_, err = ops.SPCClient.Create(spcObj)
+			Expect(err).To(BeNil(), "while creating spc", spcName)
+
+			By("verifying healthy csp count")
+			cspCount := ops.GetHealthyCSPCountEventually(spcName, cstor.PoolCount)
+			Expect(cspCount).To(Equal(true), "while checking cstorpool health status")
+
+		})
+	})
+
+	AfterEach(func() {
+		By("deleting resources created for testing cstor volume provisioning", func() {
+			By("deleting storageclass")
```

Comment:
If BeforeEach does not have StorageClass then AfterEach should not deal with StorageClass